File: imagepicker/src/main/java/com/nguyenhoanglam/imagepicker/ui/common/BasePresenter.java
package com.nguyenhoanglam.imagepicker.ui.common;

/**
 * Created by hoanglam on 8/17/17.
 */

public class BasePresenter<T extends MvpView> {

    private T view;

    public void attachView(T view) {
        this.view = view;
    }

    public T getView() {
        return view;
    }

    public void detachView() {
        view = null;
    }

    protected boolean isViewAttached() {
        return view != null;
    }

}

File: imagepicker/src/main/java/com/nguyenhoanglam/imagepicker/ui/common/BasePresenter.kt
package com.nguyenhoanglam.imagepicker.ui.common

/**
 * Created by hoanglam on 8/17/17.
 */
open class BasePresenter<T : MvpView?> {
    var view: T? = null
        private set

    fun attachView(view: T) {
        this.view = view
    }

    fun detachView() {
        view = null
    }

    protected val isViewAttached: Boolean
        protected get() = view != null
}